packages/@aws-cdk/aws-ec2-alpha/lib/route.ts

```diff
@@ -402,6 +444,88 @@ export class NatGateway extends Resource implements IRouteTarget {
   }
 }
 
+/**
+ * Creates a peering connection between two VPCs
+ * @resource AWS::EC2::VPCPeeringConnection
+ */
+export class VPCPeeringConnection extends Resource implements IRouteTarget {
+
+  /**
+   * The type of router used in the route.
+   */
+  readonly routerType: RouterType;
+
+  /**
+   * The ID of the route target.
+   */
+  readonly routerTargetId: string;
+
+  /**
+   * The VPC peering connection CFN resource.
+   */
+  public readonly resource: CfnVPCPeeringConnection;
+
+  constructor(scope: Construct, id: string, props: VPCPeeringConnectionProps) {
+    super(scope, id);
+
+    const region = props.acceptorRegion ? props.acceptorRegion : Stack.of(this).region;
+    let peerRole: Role | undefined;
+    this.routerType = RouterType.VPC_PEERING_CONNECTION;
+
+    if (props.isCrossAccount && props.acceptorAccountId === undefined) {
+      throw new Error('AcceptorAccountId is required for cross-account peering connections');
+    }
+
+    const overlap = validateVpcCidrOverlap(props.requestorVpc, props.acceptorVpc);
```

Comment:
will we be able to run this validation for imported resource, or shall we leave it to runtime?

Reply:
With new design changes, we will be able to run validation for imported resources. Confirmed with Shikha we are keeping validation here and not for runtime for faster error checking.